src/garcia_almeida.jl

```diff
@@ -242,3 +241,37 @@ function garcia_almeida_break_criterion(FNNs, counter, fnn_thres, max_num_of_cyc
     end
     return flag
 end
+
+"""
+Return the minima of the given time series s
+"""
+function get_minima(s::Vector{T}) where {T}
+    minimas = Int[]
+    N = length(s)
+    flag = false
+    first_point = 0
+    for i = 2:N-1
+        if s[i-1] > s[i] && s[i+1] > s[i]
+            flag = false
+            push!(minimas, i)
+        end
+        # handling constant values
+        if flag
+            if s[i+1] > s[first_point]
+                flag = false
+                push!(minimas, first_point)
+            elseif s[i+1] < s[first_point]
+                flag = false
+            end
+        end
+        if s[i-1] > s[i] && s[i+1] == s[i]
+            flag = true
+            first_point = i
+        end
+    end
+    # make sure there is no empty vector returned
+    if isempty(minimas)
+        _, minimas = findmin(s)
```

Comment:
`minimas` is initialized as a vector of integers at the start of the function. Here `minimas` becomes an `Int` instead of vector, which makes the return type of the function type unstable. Is the return type `Int` or `Vector{Int}`? The simple change of `_, minx = findmin(s); return [minx]` ensures that always a vector of integers is returned. 

I have a better question however. Since you only need the first minimum, why do you bother collecting all minima instead of just stopping after the first and returning the index of the first minimum? You anyways do ` τs[min_idx[1]]` in the above function...

Reply:
Hi @Datseris,

> minimas is initialized as a vector of integers at the start of the function. Here minimas becomes an Int instead of vector, which makes the return type of the function type unstable. Is the return type Int or Vector{Int}? The simple change of _, minx = findmin(s); return [minx] ensures that always a vector of integers is returned

Thank you for this valuable comment. I changed it. 

> I have a better question however. Since you only need the first minimum, why do you bother collecting all minima instead of just stopping after the first and returning the index of the first minimum? You anyways do τs[min_idx[1]] in the above function...

You are right. Since I'll need to find all maximas in a time series for my other idea regarding the Pecora-method, I set up this code, to be easily reusable for this task as well.

Reply:
> You are right. Since I'll need to find all maximas in a time series for my other idea regarding the Pecora-method, I set up this code, to be easily reusable for this task as well.

Okay, that is good idea then